main/src/main/java/org/qbicc/main/PlatformTypeSystemLoader.java

```diff
@@ -0,0 +1,213 @@
+            CProbe.Type type_id_type = smallTypeIds ? int16_t : int32_t;
+            tsBuilder.setTypeIdSize((int) probeResult.getTypeInfo(type_id_type).getSize());
+            tsBuilder.setTypeIdAlignment((int) probeResult.getTypeInfo(type_id_type).getAlign());
+            tsBuilder.setEndianness(probeResult.getByteOrder());
+            if (nogc) {
+                new NoGcTypeSystemConfigurator().accept(tsBuilder);
+            }
+
+            return tsBuilder.build();
+        }
+        return null;
+    }
+
+    TypeSystem fromYaml() throws IOException {
+        String fpath = "bundles/" + platform.toString() + "/platform-type-info.yaml";
+        InputStream stream = this.getClass().getClassLoader().getResourceAsStream(fpath);
+        if (stream == null) return null;
+        JsonNode jsonNode = new ObjectMapper(new YAMLFactory()).readTree(stream);
+        JsonNode platformTypeInfo = jsonNode.get("platform-type-info");
+
+        TypeSystem.Builder tsBuilder = TypeSystem.builder();
+
+        JsonNode boolean_t = platformTypeInfo.get("boolean");
+        tsBuilder.setBoolSize(boolean_t.get("size").asInt());
+        tsBuilder.setBoolAlignment(boolean_t.get("align").asInt());
+
+        tsBuilder.setByteBits(platformTypeInfo.get("byte-bits").asInt());
+
+        tsBuilder.setSignedChar(platformTypeInfo.get("signed-char").asBoolean());
+
+        JsonNode int8_t = platformTypeInfo.get("int8");
+        tsBuilder.setInt8Size(int8_t.get("size").asInt());
+        tsBuilder.setInt8Alignment(int8_t.get("align").asInt());
+
+        JsonNode int16_t = platformTypeInfo.get("int16");
+        tsBuilder.setInt16Size(int16_t.get("size").asInt());
+        tsBuilder.setInt16Alignment(int16_t.get("align").asInt());
+
+        JsonNode int32_t = platformTypeInfo.get("int32");
+        tsBuilder.setInt32Size(int32_t.get("size").asInt());
+        tsBuilder.setInt32Alignment(int32_t.get("align").asInt());
+
+        JsonNode int64_t = platformTypeInfo.get("int64");
+        tsBuilder.setInt64Size(int64_t.get("size").asInt());
+        tsBuilder.setInt64Alignment(int64_t.get("align").asInt());
+
+        JsonNode float32_t = platformTypeInfo.get("float32");
+        tsBuilder.setFloat32Size(float32_t.get("size").asInt());
+        tsBuilder.setFloat32Alignment(float32_t.get("align").asInt());
+
+        JsonNode float64_t = platformTypeInfo.get("float64");
+        tsBuilder.setFloat64Size(float64_t.get("size").asInt());
+        tsBuilder.setFloat64Alignment(float64_t.get("align").asInt());
+
+        JsonNode pointer_t = platformTypeInfo.get("pointer");
+        tsBuilder.setPointerSize(pointer_t.get("size").asInt());
+        tsBuilder.setPointerAlignment(pointer_t.get("align").asInt());
+
+        tsBuilder.setMaxAlignment(platformTypeInfo.get("max-alignment").asInt());
+        // todo: function alignment probe
+        // for now, references == pointers
+        JsonNode reference_t = platformTypeInfo.get("reference");
```

Comment:
ok, maybe I could add a nested static enum 🤔

Reply:
notice this is being overridden in `NoGcTypeSystemConfigurator#accept` anyway 🤔

Reply:
True, that could work as a way to specify it from the heap scheme. But in the meantime the `reference` size should not appear in the platform file IMO since it is not determined by the platform. Just mapping it to size/align of pointer is fine.